src/contracts/interfaces/IPermissionController.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+interface IPermissionControllerErrors {
+    /// @notice Thrown when the caller is not the admin
+    error NotAdmin();
+    /// @notice Thrown when an admin is set to the zero address
+    error AdminAlreadySet();
+    /// @notice Thrown when the admin to remove is not an admin
+    error AdminNotSet();
+    /// @notice Thrown when a delegate is already set for the account's function
+    error DelegateAlreadySet();
+    /// @notice Thrown when a delegate is not set for the account's function
+    error DelegateNotSet();
+}
+
+interface IPermissionControllerEvents {
+    /// @notice Emitted when a delegate is set
+    event DelegateSet(address indexed account, address indexed delegate, address target, bytes4 selector);
+
+    /// @notice Emitted when a delegate is revoked
+    event DelegateRemoved(address indexed account, address indexed delegate, address target, bytes4 selector);
+
+    /// @notice Emitted when an admin is set for a given account
+    event AdminSet(address indexed account, address admin);
+
+    /// @notice Emitted when an admin is removed for a given account
+    event AdminRemoved(address indexed account, address admin);
+}
+
+interface IPermissionController is IPermissionControllerErrors, IPermissionControllerEvents {
+    /**
+     * @notice Set the admin of an account
+     * @param account to set admin for
+     * @param admin to set
+     * @dev Multiple admins can be set for an account
+     */
+    function setAdmin(address account, address admin) external;
+
+    /**
+     * @notice Remove an admin of an account
+     * @param account to remove admin for
+     * @param admin to remove
+     * @dev Only the admin of the account can remove an admin
+     */
+    function removeAdmin(address account, address admin) external;
+
+    /**
+     * @notice Set a delegate for a given account
+     * @param account to set delegate for
+     * @param delegate to set
+     * @param target to set delegate for
+     * @param selector to set delegate for
+     * @dev Only the admin of the account can set a delegate
+     */
+    function setDelegate(address account, address delegate, address target, bytes4 selector) external;
```

Comment:
Is this getting renamed? As discussed before we should not confuse this with staker<>operator delegation.
I suggested something like `Entrustee` or `Entrant`. Entrant seems like it describes the actor who can enter specific function selectors.

Reply:
not a fan on `Entrant`

Reply:
How about agent??

Reply:
Went with appointee: https://github.com/Layr-Labs/eigenlayer-contracts/pull/870/commits/cf26f4eb29af4a89763b2e26b99802c8fd04d090